File: app/src/main/java/com/kota/Telnet/TelnetContainer.java
package com.kota.Telnet;

import java.util.Map;
import java.util.TreeMap;

public class TelnetContainer<ObjectType> {
    Map<Integer, ObjectType> _container = new TreeMap<>();
    int _max_index = 0;

    public void add(Integer index, ObjectType object) {
        if (this._container != null) {
            this._container.put(index, object);
            if (index > this._max_index) {
                this._max_index = index;
            }
        }
    }

    public ObjectType get(Integer index) {
        if (index < 0 || index > this._max_index || this._container == null || !this._container.containsKey(index)) {
            return null;
        }
        return this._container.get(index);
    }

    public int size() {
        if (this._container != null) {
            return this._max_index + 1;
        }
        return 0;
    }

    public void clear() {
        this._max_index = 0;
        this._container.clear();
    }
}

File: app/src/main/java/com/kota/telnet/TelnetContainer.kt
package com.kota.telnet

import java.util.TreeMap

class TelnetContainer<ObjectType> {
    var objectContainer: MutableMap<Int?, ObjectType?>? = TreeMap<Int?, ObjectType?>()
    var maxIndex: Int = 0

    fun add(index: Int, `object`: ObjectType?) {
        if (this.objectContainer != null) {
            this.objectContainer!!.put(index, `object`)
            if (index > this.maxIndex) {
                this.maxIndex = index
            }
        }
    }

    fun get(index: Int): ObjectType? {
        if (index < 0 || index > this.maxIndex || this.objectContainer == null || !this.objectContainer!!.containsKey(
                index
            )
        ) {
            return null
        }
        return this.objectContainer!![index]
    }

    fun size(): Int {
        if (this.objectContainer != null) {
            return this.maxIndex + 1
        }
        return 0
    }

    fun clear() {
        this.maxIndex = 0
        this.objectContainer!!.clear()
    }
}
